host/host_linux.go

```diff
@@ -322,7 +330,7 @@ func getRedhatishVersion(contents []string) string {
 	if strings.Contains(c, "rawhide") {
 		return "rawhide"
 	}
-	if matches := regexp.MustCompile(`release (\d[\d.]*)`).FindStringSubmatch(c); matches != nil {
+	if matches := regexp.MustCompile(`release (\w[\d.]*)`).FindStringSubmatch(c); matches != nil {
```

Comment:
I don't know the neokylin release string, but it's going to affect other behaviors.　Could you revert this?

(Note: I have checked original [ohai code](https://github.com/chef/ohai/blob/de7b4f4349c767120f223b0569687187082ad73c/lib/ohai/plugins/linux/platform.rb#L36) and noticed this is updated. So I am thinking about to change this regex.)

Reply:
I know that, neokylinux is one branch of redhat,  the content of **/etc/neokylin-release** is :
```
NeoKylin Linux Advanced Server release V7 (Chromium)
```
 "V7" that with string and number can match by **\w** that  with string and number, this is one generic way.

Reply:
I know that, neokylinux is one branch of redhat,  the content of **/etc/neokylin-release** is :
```
NeoKylin Linux Advanced Server release V7 (Chromium)
```
 "V7" that with string and number can match by **\w** that  with string and number, this is one generic way.

Reply:
Thank you for the info. To pass the [Regex test](https://rubular.com/r/78c1yXYa7zDhdV) from [Ohai code](https://github.com/chef/ohai/blob/de7b4f4349c767120f223b0569687187082ad73c/lib/ohai/plugins/linux/platform.rb#L29), how about to change this? (I found current implementation also does not pass the Amazon Linux 2...)
```suggestion
	if matches := regexp.MustCompile(`release (\w?[\d.]*)`).FindStringSubmatch(c); matches != nil {
```

Reply:
\w match string/number/_
? match last byte occury once or zero,
so add ? or not all is ok. do you think ?

others , I test some cases as below:
```

ss := []string{
		"NeoKylin Linux Advanced Server release 7.6.1 (Chromium)",
		"NeoKylin Linux Advanced Server release V7.6 (Chromium)",
		"NeoKylin Linux Advanced Server release v11.2.32.1 (Chromium)",
		"Amazon Linux 2",
		"Amazon Linux AMI release 2018.03",
		"Amazon Linux AMI release v2018.03",
		"CentOS Linux release 7.6.1810 (Core)",
		"CentOS release 6.10 (Final)",
		"Fedora release 29 (Twenty Nine)",
		"Red Hat Enterprise Linux release 8.0 Beta (Ootpa)",
		"Red Hat Enterprise Linux release .012 Beta (Ootpa)",
	}
	
for _, s := range ss {
        if matches := regexp.MustCompile(`release (\w[\d.]*)`).FindStringSubmatch(s); matches != nil {
	        log.Printf("%s => %s", s, matches[1])
        } else {
	        log.Printf("%s => ", s)
        }
}

#### output :
2022/07/10 16:38:41 res:2022-07-10T06:10:01.65027+08:00
2022/07/10 16:38:41 NeoKylin Linux Advanced Server release 7.6.1 (Chromium) => 7.6.1
2022/07/10 16:38:41 NeoKylin Linux Advanced Server release V7.6 (Chromium) => V7.6
2022/07/10 16:38:41 NeoKylin Linux Advanced Server release v11.2.32.1 (Chromium) => v11.2.32.1
2022/07/10 16:38:41 Amazon Linux 2 => 
2022/07/10 16:38:41 Amazon Linux AMI release 2018.03 => 2018.03
2022/07/10 16:38:41 Amazon Linux AMI release v2018.03 => v2018.03
2022/07/10 16:38:41 CentOS Linux release 7.6.1810 (Core) => 7.6.1810
2022/07/10 16:38:41 CentOS release 6.10 (Final) => 6.10
2022/07/10 16:38:41 Fedora release 29 (Twenty Nine) => 29
2022/07/10 16:38:41 Red Hat Enterprise Linux release 8.0 Beta (Ootpa) => 8.0
2022/07/10 16:38:41 Red Hat Enterprise Linux release .012 Beta (Ootpa) => 
```

Reply:
As you tested, `(\w[\d.]*)` can not capture `Amazon Linux 2`. `(\w?[\d.]*)` can capture. So I think adding `?` is better.

Reply:
as you see,  use the regex **(\w?[\d.]*)** ,
```
ss := []string{
		"NeoKylin Linux Advanced Server release 7.6.1 (Chromium)",
		"NeoKylin Linux Advanced Server release V7.6 (Chromium)",
		"NeoKylin Linux Advanced Server release v11.2.32.1 (Chromium)",
		"Amazon Linux 2",
		"Amazon Linux AMI release 2018.03",
		"Amazon Linux AMI release v2018.03",
		"CentOS Linux release 7.6.1810 (Core)",
		"CentOS release 6.10 (Final)",
		"Fedora release 29 (Twenty Nine)",
		"Red Hat Enterprise Linux release 8.0 Beta (Ootpa)",
		"Red Hat Enterprise Linux release .012 Beta (Ootpa)",
	}

	for _, s := range ss {
		if matches := regexp.MustCompile(`release (\w?[\d.]*)`).FindStringSubmatch(s); matches != nil {
			log.Printf("%s => %s", s, matches[1])
		} else {
			log.Printf("%s => ", s)
		}
	}
######## output 
2022/07/12 10:33:10 NeoKylin Linux Advanced Server release 7.6.1 (Chromium) => 7.6.1
2022/07/12 10:33:10 NeoKylin Linux Advanced Server release V7.6 (Chromium) => V7.6
2022/07/12 10:33:10 NeoKylin Linux Advanced Server release v11.2.32.1 (Chromium) => v11.2.32.1
2022/07/12 10:33:10 Amazon Linux 2 => 
2022/07/12 10:33:10 Amazon Linux AMI release 2018.03 => 2018.03
2022/07/12 10:33:10 Amazon Linux AMI release v2018.03 => v2018.03
2022/07/12 10:33:10 CentOS Linux release 7.6.1810 (Core) => 7.6.1810
2022/07/12 10:33:10 CentOS release 6.10 (Final) => 6.10
2022/07/12 10:33:10 Fedora release 29 (Twenty Nine) => 29
2022/07/12 10:33:10 Red Hat Enterprise Linux release 8.0 Beta (Ootpa) => 8.0
2022/07/12 10:33:10 Red Hat Enterprise Linux release .012 Beta (Ootpa) => .012
```
i think that use special writing to match "Amazon Linux 2" , other than `release (\w?[\d.]*)`

Reply:
Ah, sorry, regex is different from Ruby and go. You are correct. Let me add Amazon Linux 2 specific check on another PR. Thank you.